### PR TITLE
Docs/contributing plus cleanup

### PR DESCRIPTION
**Pull request by FinsaasGH**
Having an example workflow for dev vs repo admin may be helpful. The steps themselves are explained very clearly, only the responsibility of each party is unclear.

Dev: create new branch from master (GH-600 etc) -> PR completed card to master -> discuss with reviewer and/or repo admin if the merge warrants a version bump -> branch is merged into master

Repo admin: review/discuss card with dev and or reviewer -> if merge warrants version bump, create git tag with next version and instruct dev to increment cargo.tomls(if no version bump required, create a pre-release tag) -> complete merge and apply appropriate tag to merge commit

### CODE REVIEW

File: PRACTICE.md
Comment (FinsaasGH) [inline PRACTICE.md]: Do we want to have non-release candidates tagged or just leave them as normal merges into master? Ex 3 cards merge into master, not one of the cards warrants a release by itself. Once the last card is going through PR can the cargo.tomls be incremented for that branch and tag the commit for release.
This will give a cleaner repo vs having additional pre-release.x tags, while also simplifying QA within React-app using "latest" binaries. If we want to start using versioned workflows within React-app this will require non-trivial testing.

Comment (kauri-hero) [reply PRACTICE.md]: Well I guess the main reason for the tagging of non-release candidates is that its changing the master codebase each time.
so lets say something does need to be identified or rolled back (or even tested) then its easy to checkout the git tag of the master branch because every merge had a git tag, instead of searching based on commits and commit SHA.

But to be clear I didn't intend on the cargo.tomls being edited each time with the pre-release.x versions, its only for the git tag.

But to be honest this can be managed either way with clear and concise communication within the devs team and whoever is conducting the merges. Since master branch is protected it could be enforced by maintainers, which current are Dan, Bert and I.
before any merge this could be checked

File: PRACTICE.md
Comment (bertllll) [inline PRACTICE.md:209]: Why do we talk here about some general ways of the format if we later choose and talk just about our specific way. I find that a bit distracting if we put information here which we don't need to follow. I'd appreciate if it were more about us and less about all possibilities out there.

File: node/src/ui_gateway/README.md
Comment (bertllll) [inline node/src/ui_gateway/README.md]: This is a bit misleading. The referred document is not to describe UIGateway (It speaks in detail of messages traveling between certain two points.), however you mention it in a read-me originally belonging only to this actor. 

Maybe "Further documentation, or Associated documentation..."?

File: PRACTICE.md
Comment (bertllll) [inline PRACTICE.md]: I think we rather want to adjust that with the final PR within the release period. Or put differently, with the last branch merged in the release. 

We certainly don't want to change these toml files with every minor commit. 

This time, by chance, I know Dan's already changed the cargo versioning in his card, merged by now. So even though it may have been considered in fact too early it doesn't harm anything really much. We will just not have to do it anymore for this upcoming release.

File: PRACTICE.md
Comment (bertllll) [inline PRACTICE.md]: What I hoped for that I would read in this documentation is some sort of show of particular steps that you need to take in order to create the tag.

I thought a bit about it now and although I don't have very good knowledge of that I'd add something like this:

"About technical aspects of the tag. We think we need only the so-called lightweight tag, that is without an extend in the form of annotation. A possibly right way of taking care of the release tag is first to go to acquire an agreement from the product owner that your branch (in your open PR) is going to enclose the current, pending release. If this is the case you want to make sure your code in your branch appears errorless in Actions (it's passed all checks) and also it's satisfied all reviews and so is approved by the reviewer. 

At that point you create a tag to the HEAD commit of your branch with the previously agreed version number. 
If you do that locally on your machine you may want to use e.g.:

```
git tag v0.6.1
git push origin v0.6.1
```

The last step should cause a new trigger of a run in Actions, this time with a different workflow than usual, producing official release binaries at the end. 
 
Beside that, while you know the tag is in its place you're finally ready to let the merge button for your PR to master be pressed."

Maybe somewhere here I'm not accurate or even wrong in the steps so please think it over together with Finsaas. ;)

File: PRACTICE.md
Comment (kauri-hero) [reply PRACTICE.md:209]: fair comment I'll make it specific for us

File: PRACTICE.md
Comment (kauri-hero) [reply PRACTICE.md]: bert had a good conversation on this also @FinsaasGH 

we will forgo git tagging every commit, and only the commit to master that will have a release candidate.
if it needs to be a prerelease at that time, we can git tag it and have the release package labelled prerelease for testing if needed

File: PRACTICE.md
Comment (kauri-hero) [reply PRACTICE.md]: I have adjusted this and finsaas is aligned with this approach as well

File: PRACTICE.md
Comment (kauri-hero) [reply PRACTICE.md]: great I have added this with a few edits to your outline

Thanks @bertllll !

File: node/src/ui_gateway/README.md
Comment (kauri-hero) [reply node/src/ui_gateway/README.md]: I have adjusted this now - thanks

**Review by bertllll — APPROVED** [review]

**Pull request merged**
